104-max-depth-binary-tree.js

```diff
@@ -0,0 +1,49 @@
+// Challenge: https://leetcode.com/problems/maximum-depth-of-binary-tree/
+
+// Objective: Given the root of a binary tree, return its maximum depth.
+
+// Note: A binary tree's maximum depth is the number of nodes along the longest path
+// from the root node down to the farthest leaf node.
+
+// Solution:
+const assert = require("assert");
+
+/*
+ Definition for a binary tree node.
+ function TreeNode(val, left, right) {
+ 	this.val = (val===undefined ? 0 : val)
+ 	this.left = (left===undefined ? null : left)
+  this.right = (right===undefined ? null : right)
+  }
+*/
+/*
+ @param {TreeNode} root
+ @return {number}
+*/
+
+var maxDepth = function(root) {
+  var recur = function(n) {
+    // check values using strict equal
+    // reach leaf, return 0
+    if (!n) {
```

Comment:
No changes needed - just addle info. `if (!n) {}` isn't technically a strict equal comparison. an undefined variable or empty string would also cause the code inside this if block to fire. 

https://www.w3docs.com/snippets/javascript/how-to-check-for-empty-undefined-null-string-in-javascript.html

Reply:
Oh interesting. Good to know. Thanks!